Fix symlink block allocation mismatch causing unlink bitmap errors

RoyWFHuang opened the pull request:
When we create symbolic link, in simplefs_new_inode function, the symbolic
does not use ei_block to record the using block. But in simplefs_unlink function we use the ei_block to clean the bitmap, this makes the unlink will clean the "0" bit in bitmap.

Close #58

Also fixing the warning messages.
```
[13519.467953] ------------[ cut here ]------------
[13519.467954] WARNING: CPU: 1 PID: 287149 at fs/inode.c:307 drop_nlink+0x2c/0x40
....
[13519.468072]  ? drop_nlink+0x2c/0x40
[13519.468074]  ? __warn+0x8c/0x100
[13519.468077]  ? drop_nlink+0x2c/0x40
[13519.468080]  ? report_bug+0xa4/0xd0
[13519.468083]  ? handle_bug+0x39/0x90
[13519.468086]  ? exc_invalid_op+0x19/0x70
[13519.468089]  ? asm_exc_invalid_op+0x1b/0x20
[13519.468094]  ? drop_nlink+0x2c/0x40
[13519.468097]  simplefs_unlink+0x2e9/0x3b0 [simplefs]

```
the [inode_dec_link_count](https://elixir.bootlin.com/linux/v6.10.5/source/fs/inode.c#L330) has included drop_nlink and mark_inode_dirty

HotMercury commented:
This PR indeed fixes the symbolic bug. 
Could you please provide a test case in the test file?

RoyWFHuang commented:
Updated.

jserv reviewed (changes requested):
Consolidate the test suite to reflect the proposed changes as @HotMercury requested.

RoyWFHuang committed:
Fix Symbolic link remove problem

    When we create symbolic link, in simplefs_new_inode function, the symbolic
does not use ei_block to record the using block. But in simplefs_unlink functoin
we use the ei_block to clean the bitmap, this makes the unlink will clean the "0"
bit in bitmap.

Close #58

RoyWFHuang committed:
Fix warning messages in remove symbolic

inode_dec_link_count has included drop_nlink/ mark_inode_dirty

jserv commented:
Thank @RoyWFHuang for contributing!
